roles/glance/tasks/main.yml

```diff
@@ -49,23 +55,21 @@
     - glance-api
     - glance-registry
 
-- name: start glance services
-  service: name={{ item }} state=started
-  with_items:
-    - glance-api
-    - glance-registry
-
 - name: glance config
   template: src={{ item }} dest=/etc/glance mode=0644
   with_fileglob: ../templates/etc/glance/*
   notify:
     - restart glance services
 
-- name: permit access to glance
-  ufw: rule=allow to_port={{ item }} proto=tcp
+- name: sync glance database
+  command: glance-manage db_sync
+  when: database_create.changed or force_sync|default('false')|bool
+
+- name: start glance services
+  service: name={{ item }} state=started
   with_items:
-    - 9292
-    - 9293
+    - glance-api
+    - glance-registry
```

Comment:
Actually re-reading. The first time though, there /is/ no glance service to restart, until the task that installs the service file. So that handler just fails gracefully, due to restart_if_present (which incidentally we should stop using the custom module for that). So I take back my objection to the ordering here.